core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -301,7 +368,7 @@ class IndexingQueueActor(searchService: SearchService) extends Actor with ActorL
   // De-dupes files that have been updated since we were last told to
   // index them. No need to aggregate values: the latest wins. Key is
   // the URI because FileObject doesn't implement equals
-  var todo = Map.empty[String, FileObject]
+  val todo = new m.HashMap[FileObject, m.Set[FileObject]] with m.MultiMap[FileObject, FileObject]
```

Comment:
I needed a multimap so mutable container was my only option, but since the is (afaik) no default way to split multimap into multimaps, I went with `val` and simple key removal.